### PR TITLE
Integrate `Fertilizer` module into `Soil` and `Field` modules

### PR DESCRIPTION
**Pull request by Isgotkowitz**
# Main Changes
- `soil.py`
  -  Import statement for `Fertilizer` has been added, an unnecessary one has been removed. 
  - There is now a `Fertilizer` attribute to in `Soil`
  - The `make_from_config()` class method now takes field size as an argument (it is needed by methods within the `Fertilizer` module
  - `Soil` now calls the daily fertilizer phosphorus update routine as part of the `daily_soil_water_routine()`
- `soil_data.py`
  - The annual total of phosphorus leached from the field by runoff is now tracked, and is reset along with all other annual totals tracked by the `SoilData` module
  - The incorporation of `Fertilizer` into `SoilData` is tested in `test_soil_data_and_config_factory.py`
- `layer_data.py`
  - a labile phosphorus content attribute has been added to track the amount of labile phosphorus in each soil layer 
- `field.py`
  - `amend_soil()` now adds fertilizer phosphorus to the field
  - the daily update routine for soil is now called with the new method sigature 
# Other Changes
- `fertilizer.py` had a small docstring formatting issue corrected 
- Most files in this PR have had small formatting issues addressed that were raised when linting with `flake8`
- `layer_data.py` has had duplicate `@property` method removed
# Notes
- Many of the integrations of `Fertilizer` in the `Field` module are not tested, and will wait to be tested until they are merged with a branch that includes the `test_field.py` file
- Closes https://github.com/RuminantFarmSystems/MASM/issues/379

### CODE REVIEW

**Review by morrowcj — APPROVED** [review]
LGTM!

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 1]
```diff
@@ -56,11 +56,11 @@ def manage_field(self, day: int, year: int, current_weather: CurrentWeather) ->
         # determine total amount of residue and above-ground biomass present on the given day
         total_plant_cover = self.field_data.current_residue + self._determine_total_above_ground_biomass()
 
+        # TODO: track snow cover on soil surface somewhere - Issue #317
         self.soil.daily_soil_routine(current_weather.incoming_light, current_weather.mean_air_temperature,
                                      current_weather.min_air_temperature, current_weather.max_air_temperature,
```
Comment (ltp23) [inline SC_redesign/Crop_and_Soil/field/field.py:69]: This list of parameters is getting long, so maybe we should start to include parameter names as well.

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 2]
```diff
@@ -134,7 +134,11 @@ def till_soil(self) -> None:
 
     def amend_soil(self) -> None:
         """amend the soil with nutrients"""
-        pass
+        # TODO: test this method when it is merged with a branch containing field_test.py
+        fertilizer_phosphorus_applied = 90
+        if fertilizer_phosphorus_applied:
+            self.soil.fertilizer_phosphorus.add_fertilizer_phosphorus(fertilizer_phosphorus_applied)
```
Comment (ltp23) [inline SC_redesign/Crop_and_Soil/field/field.py]: Probably only this line is needed here. I think the `add_fertilizer_phosphorus` should be able to handle the case of input 0 by doing nothing.

**Review by ltp23 — APPROVED** [review]
Looks good to me! Thanks, Ed.

File: SC_redesign/Crop_and_Soil/field/field.py (hunk 1 above)
Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/field/field.py:69]: This is a good point. I've wondered when/if we need to employ keywords explicitly when calling methods.

**Review by morrowcj — APPROVED** [review]

File: SC_redesign/Crop_and_Soil/field/field.py (hunk 1 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/field/field.py:69]: I will change this up

File: SC_redesign/Crop_and_Soil/field/field.py (hunk 2 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/field/field.py]: Good point, that would be a more sensible placeholder

**Pull request merged**